app/services/AppealService.scala

```diff
@@ -123,21 +128,23 @@ class AppealService @Inject()(appealsConnector: PEGAConnector,
     }
   }
 
-  private def sanitisedAndTruncatedFileName(fileName: String)(fileMimeType: String)(reference: String): String = {
+  def sanitisedAndTruncatedFileName(fileName: String)(fileMimeType: String)(reference: String): String = {
     val sanitisedFileName = sanitiseFileName(fileName)(fileMimeType)(reference)
     if (sanitisedFileName.length > appConfig.maximumFilenameLength) {
-      if (sanitisedFileName.contains(".")) {
-        val fileRegex = "^(.*)(\\.\\w{1,4})$".r
-        val fileRegex(fileNameMain, fileExtension) = sanitisedFileName
-        logger.info(s"[AppealService][sanitisedAndTruncatedFileName] File name length: ${fileNameMain.length} with reference of: $reference, truncating to ${appConfig.maximumFilenameLength}")
-        fileNameMain.substring(0, Math.min(fileNameMain.length(), appConfig.maximumFilenameLength)) ++ fileExtension
-      } else {
-        logger.info(s"[AppealService][sanitisedAndTruncatedFileName] File name length: ${sanitisedFileName.length} with reference of: $reference, truncating to ${appConfig.maximumFilenameLength}")
-        sanitisedFileName.substring(0, Math.min(sanitisedFileName.length(), appConfig.maximumFilenameLength))
+      val fileWithExtensionRegex = "^(.*)(\\.\\w{1,4})$".r
+      sanitisedFileName match {
+        case fileWithExtensionRegex(fileNameMain, fileExtension) =>
+          logger.info(s"[AppealService][sanitisedAndTruncatedFileName] File name length: ${fileNameMain.length} " +
+            s"with reference of: $reference, truncating to ${appConfig.maximumFilenameLength}")
+          fileNameMain.substring(0, Math.min(fileNameMain.length(), appConfig.maximumFilenameLength)) ++ fileExtension
+
+        case _ =>
+          logger.info(s"[AppealService][sanitisedAndTruncatedFileName] File name length: ${sanitisedFileName.length} " +
+            s"with reference of: $reference, truncating to ${appConfig.maximumFilenameLength}")
+          sanitisedFileName.substring(0, Math.min(sanitisedFileName.length(), appConfig.maximumFilenameLength))
```

Comment:
This addresses a warning that 'regex search isn't exhaustive'.
The regex takes a string with an extension and returns the filename and the extension separately, i.e.
`myFilename.txt -> ('myFilename', '.txt')`

Before, the `val fileRegex(fileNameMain, fileExtension) = sanitisedFileName` would only work if the file name contained a '.' which should always be the case because it's inside the `if (sanitisedFileName.contains(".")) {` block.
However, I didn't want to suppress the warning and this makes it safer in case someone edits the code and didn't realise. 
It will now match the regex case if it's possible, otherwise it will truncate without having to worry about the extension.

I created tests in AppealServiceSpec to ensure it is still working as it should.